File: generated-sources/java-undertow-server/mojang-sessions/src/main/java/com/github/asyncmc/mojang/sessions/java/undertow/server/model/PlayerSkinMetadata.java
package com.github.asyncmc.mojang.sessions.java.undertow.server.model;

import java.util.Objects;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonValue;
import io.swagger.annotations.ApiModel;
import io.swagger.annotations.ApiModelProperty;



/**
 * Contains details about the skin
 **/

@ApiModel(description = "Contains details about the skin")
@javax.annotation.Generated(value = "org.openapitools.codegen.languages.JavaUndertowServerCodegen", date = "2020-06-06T04:59:34.437175500-03:00[America/Sao_Paulo]")
public class PlayerSkinMetadata   {
  


  public enum ModelEnum {
    SLIM("slim");

    private String value;

    ModelEnum(String value) {
      this.value = value;
    }

    @Override
    @JsonValue
    public String toString() {
      return value;
    }
  }

  private ModelEnum model = ;

  /**
   * The player model, currently only \"slim\" (Alex) is valid, for Steve's model this property must be absent.
   **/
  public PlayerSkinMetadata model(ModelEnum model) {
    this.model = model;
    return this;
  }

  
  @ApiModelProperty(value = "The player model, currently only \"slim\" (Alex) is valid, for Steve's model this property must be absent.")
  @JsonProperty("model")
  public ModelEnum getModel() {
    return model;
  }
  public void setModel(ModelEnum model) {
    this.model = model;
  }


  @Override
  public boolean equals(java.lang.Object o) {
    if (this == o) {
      return true;
    }
    if (o == null || getClass() != o.getClass()) {
      return false;
    }
    PlayerSkinMetadata playerSkinMetadata = (PlayerSkinMetadata) o;
    return Objects.equals(model, playerSkinMetadata.model);
  }

  @Override
  public int hashCode() {
    return Objects.hash(model);
  }

  @Override
  public String toString() {
    StringBuilder sb = new StringBuilder();
    sb.append("class PlayerSkinMetadata {\n");
    
    sb.append("    model: ").append(toIndentedString(model)).append("\n");
    sb.append("}");
    return sb.toString();
  }

  /**
   * Convert the given object to string with each line indented by 4 spaces
   * (except the first line).
   */
  private String toIndentedString(java.lang.Object o) {
    if (o == null) {
      return "null";
    }
    return o.toString().replace("\n", "\n    ");
  }
}

File: generated-sources/dart/mojang-sessions/com/github/asyncmc/mojang/sessions/dart/model/player_skin_metadata.dart
part of openapi.api;

class PlayerSkinMetadata {
  /* The player model, currently only \"slim\" (Alex) is valid, for Steve's model this property must be absent. */
  String model = null;
  //enum modelEnum {  slim,  };{
  PlayerSkinMetadata();

  @override
  String toString() {
    return 'PlayerSkinMetadata[model=$model, ]';
  }

  PlayerSkinMetadata.fromJson(Map<String, dynamic> json) {
    if (json == null) return;
    model = json['model'];
  }

  Map<String, dynamic> toJson() {
    return {
      'model': model
    };
  }

  static List<PlayerSkinMetadata> listFromJson(List<dynamic> json) {
    return json == null ? new List<PlayerSkinMetadata>() : json.map((value) => new PlayerSkinMetadata.fromJson(value)).toList();
  }

  static Map<String, PlayerSkinMetadata> mapFromJson(Map<String, dynamic> json) {
    var map = new Map<String, PlayerSkinMetadata>();
    if (json != null && json.length > 0) {
      json.forEach((String key, dynamic value) => map[key] = new PlayerSkinMetadata.fromJson(value));
    }
    return map;
  }
}

